pkg/reconciler/shared/tektonconfig/upgrade/post_upgrade.go

```diff
@@ -108,3 +113,71 @@ func removeClusterTaskInstallerSets(ctx context.Context, logger *zap.SugaredLogg
 	}
 	return nil
 }
+
+// removeVersionedTaskInstallerSets removes the versioned resolver tasks installersets except latest 2 versions
+func removeVersionedTaskInstallerSets(ctx context.Context, logger *zap.SugaredLogger, k8sClient kubernetes.Interface, operatorClient versioned.Interface, restConfig *rest.Config) error {
+
+	if !v1alpha1.IsOpenShiftPlatform() {
+		return nil
+	}
+
+	taskInstallerSetsLabelSelector := metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			v1alpha1.InstallerSetType: fmt.Sprintf("%s-%s", "custom", "versionedresolvertask"),
+		},
+	}
+	taskInstallerSetsLabel, err := common.LabelSelector(taskInstallerSetsLabelSelector)
+	if err != nil {
+		return err
+	}
+
+	return findAndDeleteInstallerSetsByLabelName(ctx, logger, operatorClient, taskInstallerSetsLabel)
+}
+
+// removeVersionedStepActionsInstallerSets removes the versioned resolver step actions installersets except latest 2 versions
+func removeVersionedStepActionsInstallerSets(ctx context.Context, logger *zap.SugaredLogger, k8sClient kubernetes.Interface, operatorClient versioned.Interface, restConfig *rest.Config) error {
+
+	if !v1alpha1.IsOpenShiftPlatform() {
+		return nil
+	}
+
+	stepActionsInstallerSetsLabelSelector := metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			v1alpha1.InstallerSetType: fmt.Sprintf("%s-%s", "custom", "versionedresolverstepaction"),
+		},
+	}
+	stepActionsInstallerSetsLabel, err := common.LabelSelector(stepActionsInstallerSetsLabelSelector)
+	if err != nil {
+		return err
+	}
+
+	return findAndDeleteInstallerSetsByLabelName(ctx, logger, operatorClient, stepActionsInstallerSetsLabel)
+}
+
+func findAndDeleteInstallerSetsByLabelName(ctx context.Context, logger *zap.SugaredLogger, operatorClient versioned.Interface, installerSetsLabel string) error {
```

Comment:
Can we use logger from context ?

Reply:
@PuneetPunamiya `logger` taken from context and named as `upgrade`, used across upgrade functions
* https://github.com/tektoncd/operator/blob/release-v0.74.x/pkg/reconciler/shared/tektonconfig/upgrade/upgrade.go#L74-L75
* https://github.com/tektoncd/operator/blob/release-v0.74.x/pkg/reconciler/shared/tektonconfig/upgrade/upgrade.go#L100

Reply:
Ohk, make sense in that case, merging the pr 🤙🏻